File: Task 3.1/MainActivity.java
package com.trevin.quizapp;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity {

    private EditText userName;
    private Button startQuiz;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        userName = findViewById(R.id.editNameText);
        startQuiz = findViewById(R.id.startButton);

        startQuiz.setOnClickListener(view -> {
            String name = userName.getText().toString().trim();
            if (!name.isEmpty()){
                Intent intent = new Intent(this, QuizActivity.class);
                intent.putExtra("userName", name);
                startActivity(intent);
            } else {
                Snackbar.make(view, "Please enter a name!", Snackbar.LENGTH_SHORT)
                        .setTextColor(Color.RED)
                        .setAnchorView(view)
                        .show();
            }
        });
    }
}

File: Task 5.1/MyPlaylist/ui/MainActivity.kt
package com.trevin.myplaylists.ui

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.trevin.myplaylists.R
import com.trevin.myplaylists.data.UserRepo
import com.trevin.myplaylists.data.database.PlaylistDatabase
import com.trevin.myplaylists.databinding.ActivityMainBinding
import com.trevin.myplaylists.ui.fragments.LoginFragment
import com.trevin.myplaylists.ui.fragments.SignupFragment

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    lateinit var userRepo: UserRepo

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
        val database = PlaylistDatabase.getInstance(applicationContext)
        userRepo = UserRepo.getInstance(database)

        binding.buttonLogin.setOnClickListener {
            LoginFragment().show(
                supportFragmentManager,
                "LOGIN_SHEET"
            )
        }
        binding.buttonSignup.setOnClickListener {
            SignupFragment().show(
                supportFragmentManager,
                "SIGNUP_SHEET"
            )
        }

    }

}